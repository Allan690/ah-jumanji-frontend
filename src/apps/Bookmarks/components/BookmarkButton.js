import React, { Component } from 'react';
import {connect} from 'react-redux';
import PropTypes from 'prop-types';
import {bindActionCreators} from 'redux';

import * as bookmarksActions from '../actions/bookmarksActions';

class BookmarkButton extends Component {
  constructor(props){
    super(props);
    this.handleClick = this.handleClick.bind(this);
  }

  handleClick(e){
    const bookmarkData = {
      "bookmark": true,
      "slug": e.target.id
    };

    const { actions } = this.props;
    actions.postBookmark(bookmarkData);
  }

  render() {
    const { slug, bookmarked, bookmark } = this.props;
     let classIcon;
     let labelBookmark;
     if (bookmarked) {
       if (bookmark.bookmarks!==undefined) {
         if (bookmark.bookmarks.message =="Bookmark successfully made.") {
           classIcon = "fas fa-bookmark fa-active";
           labelBookmark = "Unbookmark";
         } else {
           classIcon = "far fa-bookmark";
           labelBookmark = "Bookmark";
         }
       } else {
         classIcon = "fas fa-bookmark fa-active";
         labelBookmark = "Unbookmark";
       }

     } else {
       if(bookmark === undefined){
         //Do nothing
         classIcon = "far fa-bookmark";
         labelBookmark = "Bookmark";
       }else{
        if (bookmark.bookmarks!==undefined) {
          if (bookmark.bookmarks.message =="Bookmark successfully made.") {
            classIcon = "fas fa-bookmark fa-active";
            labelBookmark = "Unbookmark";
          } else {
            classIcon = "far fa-bookmark";
            labelBookmark = "Bookmark";
          }
        } else {
          classIcon = "far fa-bookmark";
          labelBookmark = "Bookmark";
        }
       }
     }

    return (
      <button className="dropdown-item more btn-link" type="button" id={slug} onClick={this.handleClick}>
        <i className={classIcon}  />
        &nbsp;
        {labelBookmark}
      </button>
    );
  }

}

BookmarkButton.propTypes = {
  actions: PropTypes.object.isRequired,
  slug: PropTypes.string.isRequired,
  bookmarked: PropTypes.bool.isRequired,
  bookmark: PropTypes.object.isRequired,
};

function mapStateToProps(state) {
  return {
    bookmark: state.Bookmarks
  };
}

function mapDispatchToProps(dispatch) {
  return {
    actions: bindActionCreators(bookmarksActions, dispatch)
  };
}

export default connect(mapStateToProps, mapDispatchToProps)(BookmarkButton);
