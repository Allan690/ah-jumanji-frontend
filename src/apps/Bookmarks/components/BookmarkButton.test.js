import React from 'react';

import Enzyme, { shallow } from 'enzyme';

import Adapter from 'enzyme-adapter-react-16';

import BookmarkButton from './BookmarkButton';

Enzyme.configure({ adapter: new Adapter() });


describe('<BookmarkButton />', () => {
  it('should match the snapshot', () => {
    const component = shallow(<BookmarkButton debug />);
    expect(component).toMatchSnapshot();
  });
});
