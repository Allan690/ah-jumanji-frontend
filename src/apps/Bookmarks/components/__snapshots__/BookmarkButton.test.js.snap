// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BookmarkButton /> should match the snapshot 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;
